tests/components/PopoverComponent.spec.tsx

```diff
@@ -0,0 +1,161 @@
+    }).not.toThrow();
+  });
+
+  describe('Children propTypes', () => {
+    it('should not throw when redering a PopoverComponent with only one children', () => {
+      expect(() => {
+        shallow(
+          <PopoverComponent
+            {...popoverComponentProps}
+          >
+            <span>Toggle</span>
+          </PopoverComponent>
+        );
+      }).not.toThrow();
+    });
+
+    it('should not throw when redering a PopoverComponent without childrens', () => {
+      expect(() => {
+        shallow(
+          <PopoverComponent
+            {...popoverComponentProps}
+          />
+        );
+      }).not.toThrow();
+    });
+
+    it('should not throw when redering a PopoverComponent with a boolean as second child', () => {
+      expect(() => {
+        shallow(
+          <PopoverComponent
+            {...popoverComponentProps}
+          >
+            <span>Toggle</span>
+            {false}
+          </PopoverComponent>
+        );
+      }).not.toThrow();
+    });
+  });
+
+  describe('Document click listener', () => {
+    beforeEach(() => {
+      popoverComponentWrapper = mount(
+        <div>
+          <span id='RandomElement'>Test</span>
+          <PopoverComponent
+            {...popoverComponentProps}
+          >
+            <span id='PopoverToggle'>Toggle</span>
+            <span id='PopoverElement'>Tether element</span>
+          </PopoverComponent>
+        </div>,
+        {attachTo: document.getElementById('App')}
+      );
+    });
+
+    afterEach(() => {
+      popoverComponentWrapper.unmount();
+      popoverComponentWrapper.detach();
+    });
+
+    it('should not call toggleOpenedTetherElement on click tether toggle', () => {
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(0);
+
+      $('#PopoverToggle').click();
+
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(0);
+    });
+
+    it('should not call toggleOpenedTetherElement on click tether element', () => {
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(0);
+
+      $('#PopoverElement').click();
+
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(0);
+    });
+
+    it('should call toggleOpenedTetherElement when clicking outside PopoverComponent', () => {
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(0);
+
+      $('#RandomElement').click();
+
+      expect(toggleOpenedTetherElementSpy.calls.count()).toBe(1);
+    });
+  });
+
+  describe('Document click handler without PopoverElement in DOM', () => {
+    beforeEach(() => {
+      popoverComponentWrapper = mount(
+        <div>
+          <span id='RandomElement'>Test</span>
+          <PopoverComponent
+            {...popoverComponentProps}
+          >
+            <span id='PopoverToggle'>Toggle</span>
+            {false}
+          </PopoverComponent>
+        </div>,
+        {attachTo: document.getElementById('App')}
+      );
+    });
+
+    it('should not explode', () => {
```

Comment:
Like said in the describe: Document click handler without PopoverElement in DOM

In the test before, the popover had a tether element, but here it doesn't. That's a small difference in the code that we need to cover.